src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -120,26 +124,12 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
 
         serializerOptions.MakeReadOnly();
 
-        var schemaNode = (JsonObject)serializerOptions.GetJsonSchemaAsNode(typeof(T), new()
-        {
-            TreatNullObliviousAsNonNullable = true,
-            TransformSchemaNode = static (context, node) =>
-            {
-                if (node is JsonObject obj)
-                {
-                    if (obj.TryGetPropertyValue("enum", out _)
-                        && !obj.TryGetPropertyValue("type", out _))
-                    {
-                        obj.Insert(0, "type", "string");
-                    }
-                }
+        var schemaNode = JsonFunctionCallUtilities.InferJsonSchema(
+            typeof(T),
+            serializerOptions,
+            inferenceOptions: _inferenceOptions);
 
-                return node;
-            },
-        });
-        schemaNode.Insert(0, "$schema", "https://json-schema.org/draft/2020-12/schema");
-        schemaNode.Add("additionalProperties", false);
-        var schema = JsonSerializer.Serialize(schemaNode, JsonDefaults.Options.GetTypeInfo(typeof(JsonNode)));
+        var schema = JsonSerializer.Serialize(schemaNode, JsonDefaults.Options.GetTypeInfo(typeof(JsonElement)));
```

Comment:
Similar question here... this doesn't need to use the supplied JSO?

Reply:
Good question. In this case the schema is being rendered as a string so one could argue that indentation concerns are significant. We could, although again this would be introducing an implicit dependency for `JsonNode` on the JSO which I would rather not be enforcing on the caller.